velox/exec/Task.cpp

```diff
@@ -980,6 +989,19 @@ void Task::terminate(TaskState terminalState) {
   for (auto& bridge : oldBridges) {
     bridge->cancel();
   }
+
+  std::lock_guard<std::mutex> l(mutex_);
+  return makeFinishFutureLocked("Task::terminate");
+}
+ContinueFuture Task::makeFinishFutureLocked(const char* comment) {
+  auto [promise, future] = makeVeloxPromiseContract<bool>(comment);
+
+  if (numThreads_ == 0) {
+    promise.setValue(true);
+    return std::move(future);
+  }
+  pausePromises_.push_back(std::move(promise));
```

Comment:
I'm not sure I understand why is it OK to re-use pausePromises_ here. Would you clarify ? Should we update the comments for requestPaused that the future may complete if task is paused or have completed with some other state (success or failure)?

Reply:
Looks like some comments got lost. Repeating.

---
I'm not sure I understand why is it OK to re-use pausePromises_ here. Would you clarify ? Should we update the comments for requestPaused that the future may complete if task is paused or have completed with some other state (success or failure)?